File: c++/Queue/src/queue.cpp
#include "queue.hpp"

Queue::Queue() : head(nullptr), tail(nullptr), element_numbers(0) {

}

Queue::~Queue() {

}

void Queue::add(const int &data) {  
  if(this->tail == nullptr) {
    this->tail = new Node(data);
  }else {
    this->tail->next = new Node(data);
    this->tail = this->tail->next;
  }
  if(this->head == nullptr) {
      this->head = this->tail;
  }
  this->element_numbers++;
}

int Queue::remove() {
  int data = this->head->data;
  Node *trash = this->head; 
  this->head = this->head->next;
  if(this->head == nullptr) {
      this->tail = nullptr;
  }
  delete(trash);
  this->element_numbers--;
  return data;
}

bool Queue::contains(const int &data) {
    Node *current = this->head;
    while(current->next != nullptr) {
        if(current->data == data) {
            return true;
        }
        current = current->next;
    }
    return false;
}

int Queue::size() {
    return this->element_numbers;
}

File: python/Data Structures/Queue/queue.py
class Queue:
    class Node:
        def __init__(self, item):
            self.item = item
            self.next = None


    def __init__(self):
        self.head = None
        self.size = 0

    def add(self, item):
        if(self.head is None):
            self.head = self.Node(item)
        else:
            current = self.head
            while(current.next is not None):
                current = current.next
            current.next = self.Node(item)
        self.size += 1

    def get(self):
        if(self.head is not None):
            item = self.head.item
            self.head = self.head.next
            self.size -= 1
        return item

    def show(self):
        current = self.head
        print(current.item)
        while(current.next is not None):
            print(current.next.item)
            current = current.next

if __name__ == '__main__':
    queue = Queue()
    queue.add(2)
    queue.add(4)
    queue.add(6)
    queue.add(8)
    queue.add(10)
    queue.show()
    print(queue.get())
    print("##################")
    queue.show()